lib/registry.js

```diff
@@ -16,7 +17,9 @@ export function define(id, spec) {
   if (get(id)) {
     console.warn(`Overwriting spec ${getName(id)}`)
   }
-  spec.name = getName(id)
+  if (!spec.name && typeof spec !== "symbol") {
```

Comment:
A couple of things:

* Out of curiosity, can we use js.spec's predicates here or do we run into circular dependencies?
* You have to account for string aliases, ie `define(mySpecSymbol, "stringName")`
* I would just overwrite the name no matter what for now. We can be careful later when it's explicitly settable by the user.